contrib/google.golang.org/grpc/codec_test.go

```diff
@@ -0,0 +1,68 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016 Datadog, Inc.
+
+package grpc
+
+import (
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/mocktracer"
+	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace/tracer"
+)
+
+func TestTraceEncoding(t *testing.T) {
+	unregister := registerCodec()
+	defer unregister()
+
+	rig, err := newRig(true, WithServiceName("grpc"), WithRequestTags())
+	require.NoError(t, err, "error setting up rig")
+	defer func() { assert.NoError(t, rig.Close()) }()
+	client := rig.client
+
+	mt := mocktracer.Start()
+	defer mt.Stop()
+
+	span, ctx := tracer.StartSpanFromContext(context.Background(), "root")
+	_, err = client.Ping(ctx, &FixtureRequest{Name: "pass"})
+	require.NoError(t, err)
+	span.Finish()
+
+	spans := mt.FinishedSpans()
+	require.Len(t, spans, 7)
+
+	var (
+		clientMarshalSpan   = spans[0]
+		clientUnmarshalSpan = spans[1]
+		serverSpan          = spans[2]
+		serverMarshalSpan   = spans[3]
+		serverUnmarshalSpan = spans[4]
+		clientSpan          = spans[5]
+		rootSpan            = spans[6]
+	)
```

Comment:
Is this order always stable? It looks like a potential flaky test.

Reply:
I think it should be, the sequence of the operations should be this. The existing tests assume the order of the server / client spans and don't seem to be flaky, but if it turns out to be flaky we can change this to find the spans by name.